### DISCUSSION TITLE
About: Round-Trip Formatting，I can't understand！！

### DISCUSSION THREAD

**Discussion by Leofric-S25** [IntelliTect/EssentialCSharp.Web#521] [Feedback and Ideas]
I tried running this code：

> Listing 2.9: Formatting Using the R Format Specifier
> // ...
> const double number = 1.618033988749895;
> double result;
> string text;
>  
> text = $"{number}";
> result = double.Parse(text);
> Console.WriteLine($"{result == number}: {result} == {number}");
>  
> text = $"{number:R}";
> result = double.Parse(text);
> Console.WriteLine($"{result == number}: {result} == {number}");
>  
> // ...

The result is:
   True: result == number
   True: result == number
   True: 1.618033988749895 == 1.618033988749895
   True: 1.618033988749895 == 1.618033988749895

This result is not the same as prefetching, and I can't understand such a result.

**Comment by BenjaminMichaelis** [comment]
Hi @BigFish3710; Thanks for opening a discussion on this; there are two issues here:
1. There is a bug in the output; I am working on fixing this in the text and will push updates to the website; it should not just show "result == number"
2. The behavior of round-trip format has changed; I am looking into this but wanted to make sure you were being kept up to date. (The change was likely caused by https://github.com/dotnet/coreclr/pull/22040 if interested).

**Reply by Leofric-S25** [reply]
Thank you! For me as a beginner, I can only understand this problem now!!!
